src/upload.rs

```diff
@@ -316,9 +320,10 @@ fn complete_registry(opt: &PublishOpt) -> Result<Registry> {
     let pypirc = load_pypirc();
     let (registry_name, registry_url) = if let Some(repository_url) = opt.repository_url.as_deref()
     {
-        let name = match repository_url {
-            PublishOpt::DEFAULT_REPOSITORY_URL => Some("pypi"),
-            PublishOpt::TEST_REPOSITORY_URL => Some("testpypi"),
+        let normalized_url = PublishOpt::normalize_url(repository_url);
+        let name = match normalized_url {
+            normalized if normalized == PublishOpt::DEFAULT_REPOSITORY_URL => Some("pypi"),
+            normalized if normalized == PublishOpt::TEST_REPOSITORY_URL => Some("testpypi"),
```

Comment:
Now the `if`s can be rewriten back to simple `match` arms I think.

Reply:
Certainly, let's simplify this way.